packages/mui-joy/src/Stepper/Stepper.tsx

```diff
@@ -0,0 +1,202 @@
+'use client';
+import * as React from 'react';
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import { unstable_composeClasses as composeClasses } from '@mui/base';
+import { OverridableComponent } from '@mui/types';
+import { unstable_capitalize as capitalize } from '@mui/utils';
+import { useThemeProps } from '../styles';
+import styled from '../styles/styled';
+import { getStepperUtilityClass } from './stepperClasses';
+import { StepperProps, StepperOwnerState, StepperTypeMap } from './StepperProps';
+import useSlot from '../utils/useSlot';
+
+const useUtilityClasses = (ownerState: StepperOwnerState) => {
+  const { size, orientation } = ownerState;
+
+  const slots = {
+    root: ['root', orientation, size && `size${capitalize(size)}`],
+  };
+
+  return composeClasses(slots, getStepperUtilityClass, {});
+};
+
+const StepperRoot = styled('ol', {
+  name: 'JoyStepper',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})<{ ownerState: StepperOwnerState }>(({ theme }) => {
+  return {
+    '--Stepper-indicatorColumn': 'auto', // For vertical stepper, to control the column width of the indicator.
+    '--Step-connectorThickness': '1px',
+    '--Step-indicatorDotSize': '0.375rem',
+    variants: [
+      {
+        props: { size: 'sm' },
+        style: {
+          '--Stepper-verticalGap': '0.5rem',
+          '--Step-gap': '0.375rem',
+          '--Step-connectorInset': '0.25rem',
+          '--StepIndicator-size': '1.25rem',
+          ...theme.typography['title-sm'],
+        },
+      },
+      {
+        props: { size: 'md' },
+        style: {
+          '--Stepper-verticalGap': '0.75rem',
+          '--Step-gap': '0.5rem',
+          '--Step-connectorInset': '0.375rem',
+          '--StepIndicator-size': '1.5rem',
+          ...theme.typography['title-md'],
+        },
+      },
+      {
+        props: { size: 'lg' },
+        style: {
+          '--Stepper-verticalGap': '0.75rem',
+          '--Step-gap': '0.5rem',
+          '--Step-connectorInset': '0.5rem',
+          '--StepIndicator-size': '1.75rem',
+          ...theme.typography['title-lg'],
+        },
+      },
+      {
+        props: { orientation: 'vertical' },
+        style: { flexDirection: 'column', gap: 'var(--Stepper-verticalGap)' },
+      },
+      {
+        props: {},
+        style: {
+          boxSizing: 'border-box',
+          display: 'flex',
+          margin: 0, // Reset browser default style.
+          padding: 0, // Reset browser default style.
```

Comment:
These should go together with the variables declaration at the top (outside of the variants array).

Reply:
One more comment, if we do want to use the variants like this, shouldn't this be defined first so that it doesn't override other variant's styles? cc @brijeshb42

Reply:
I resolved this by moving the styles to the top of the definition